tests/phpunit/tests/post/nav-menu.php

```diff
@@ -265,6 +265,90 @@ public function test_update_menu_item_cache_primes_terms() {
 		$this->assertSameSets( array( $term_id ), $last[1], '_prime_term_caches() was not executed.' );
 	}
 
+
+	/**
+	 * @ticket 55620
+	 * @covers ::update_menu_item_cache
+	 */
+	public function test_wp_get_nav_menu_items_cache_primes_posts() {
+		$post_ids     = self::factory()->post->create_many( 3 );
+		$menu_nav_ids = array();
+		foreach ( $post_ids as $post_id ) {
+			$menu_nav_ids[] = wp_update_nav_menu_item(
+				$this->menu_id,
+				0,
+				array(
+					'menu-item-type'      => 'post_type',
+					'menu-item-object'    => 'post',
+					'menu-item-object-id' => $post_id,
+					'menu-item-status'    => 'publish',
+				)
+			);
+		}
+
+		// Delete post and post meta caches.
+		wp_cache_delete_multiple( $menu_nav_ids, 'posts' );
+		wp_cache_delete_multiple( $menu_nav_ids, 'post_meta' );
+		wp_cache_delete_multiple( $post_ids, 'posts' );
+		wp_cache_delete_multiple( $post_ids, 'post_meta' );
+
+		$action = new MockAction();
+		add_filter( 'update_post_metadata_cache', array( $action, 'filter' ), 10, 2 );
+
+		$num_queries = get_num_queries();
+		wp_get_nav_menu_items( $this->menu_id );
+
+		$this->assertSame( $num_queries + 7, get_num_queries(), 'Only does 7 database queries when running wp_get_nav_menu_items.' );
```

Comment:
```suggestion
		$start_num_queries = get_num_queries();
		wp_get_nav_menu_items( $this->menu_id );
		$queries_made = get_num_queries() - $start_num_queries;

		$this->assertSame( 7, $queries_made, 'Only does 7 database queries when running wp_get_nav_menu_items.' );
```

When running on trunk I get `Failed asserting that 532 is identical to 530.` which is much less helpful than `Failed asserting that 9 is identical to 7.`

:1234: This same change applies below.